src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,170 @@
+package duke.task;
+
+import duke.util.DukeException;
+import duke.util.Storage;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/**
+ * Handles the data structure where the tasks are stored.
+ * Contains various methods to add, delete and "done" a task.
+ *
+ * @author  Hardy Shein
+ * @version 0.1
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskList;
+
+    /**
+     * TaskList constructor initializes array list of tasks
+     * with initial capacity 100.
+     */
+    public TaskList() {
+        taskList = new ArrayList<>(100);
+    }
+
+    /**
+     * Getter for a reference to the task list data structure.
+     * @return the ArrayList collection of tasks.
+     */
+    public ArrayList<Task> getList() {
+        return taskList;
+    }
+
+    /**
+     * Adds to the task collection a task object generated
+     * from the save-string representation.
+     * @param taskString of a save-string representation of the task.
+     */
+    public void addSaveStringAsTask(String taskString) {
+        // Takes in a string representation of a task and adds to list
+        String taskInfo;
+        String[] sep;
+        switch (taskString.charAt(0)) {
+        case 'T':
+            taskList.add(new ToDo(taskString.substring(2, taskString.length())));
+            break;
+        case 'D':
```

Comment:
Could have one method for each case, which may help in code readability and maintenance if needed (such as for projects in the long run).